src/main/java/vazkii/botania/api/lexicon/multiblock/Multiblock.java

```diff
@@ -172,17 +202,17 @@ public void updateLocationCache() {
 	 */
 	private void addComponentToLocationCache(MultiblockComponent comp) {
 		ChunkCoordinates pos = comp.getRelativePosition();
-		locationCache.put(Arrays.asList(
-				pos.posX,
-				pos.posY,
-				pos.posZ
-				),  comp);
+		locationCache.put(hashCoordinates(pos.posX, pos.posY, pos.posZ), comp);
 	}
 
 	/**
 	 * Gets the component for a given location
 	 */
 	public MultiblockComponent getComponentForLocation(int x, int y, int z) {
-		return locationCache.get(Arrays.asList(x, y, z));
+		return locationCache.get(hashCoordinates(x, y, z));
+	}
+
+	private int hashCoordinates(int x, int y, int z) {
+		return (x * 31 + y) * 31 + z;
```

Comment:
This is going to cause collissions as soon as a multiblock exceeds 30 blocks in a dimension. While this would be safe for the vanilla Botania multis, the ones currently added by GTNH are larger already. This would effectively put an undocumented limit on their size and will probably prevent qualifying multiblocks from working with the lexicon and the associated in-world placement previews. And I would hate to see someone unaware of that having to debug related bugs in the future.

You can revert this to the original version, or use an struct-like object with the appropriate hashCode/equals implementation.